### PR TITLE
ci: add build and test with Go 1.19

### PR DESCRIPTION
**Pull request by tadelesh**
Resolve: https://github.com/Azure/azure-sdk-for-go/issues/18783

### CODE REVIEW

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
       }
     },
     "GoVersion": [
-      "1.18.2"
+      "1.18.2",
```
Comment (RickWinter) [inline eng/pipelines/templates/stages/platform-matrix.json:14]:
```suggestion
      "1.18.5",
```

**Review by RickWinter — CHANGES_REQUESTED** [review]

**Comment by jhendrixMSFT** [comment]
Superseded by https://github.com/Azure/azure-sdk-for-go/pull/18805

**Pull request closed without merging**